pkg/configuration/parameters.go

```diff
@@ -0,0 +1,156 @@
+
+The package also defines a wide array of constants representing the flags name in the command line.
+*/
+package configuration
+
+import (
+	"flag"
+	"github.com/TurnsCoffeeIntoScripts/jira-api-resource/pkg/log"
+	"strings"
+)
+
+// Definition of constants that are use for the flags setup
+const (
+	// Parameters
+	jiraAPIURL       = "url"
+	username         = "username"
+	password         = "password"
+	context          = "context"
+	issueList        = "issues"
+	customFieldName  = "customFieldName"
+	customFieldType  = "customFieldType"
+	customFieldValue = "customFieldValue"
+	loggingLevel     = "loggingLevel"
+
+	// Flags
+	secured = "secured"
+
+	// Default values and descriptions for both paramaters and flags
+	jiraAPIURLDefault           = ""
+	jiraAPIURLDescription       = "The base URL of the Jira API"
+	usernameDefault             = ""
+	usernameDescription         = "The username used to connect to the Jira API"
+	passwordDefault             = ""
+	passwordDescription         = "The password needed to connect to the Jira API"
+	contextDefault              = ""
+	contextDescription          = "The context of execution. {'EditCustomField'}"
+	issueListDefault            = ""
+	issueListDescription        = "The issue or list of issues to execute the specified context to"
+	customFieldNameDefault      = ""
+	customFieldNameDescription  = "Certain operation (such as edits) might require the user to specify the name of the custome field so that the resource may find the appropriate custom field"
+	customFieldTypeDefault      = "string"
+	customFieldTypeDescription  = "The type that is required by the field via the Jira API"
+	customFieldValueDefault     = ""
+	customFieldValueDescription = "The value of the field that will be updated (in case of update workflow)"
+	_                           = /*securedDefault*/ true
+	securedDescription          = "Flags that indicates if the API calls should be made in session"
+	loggingLevelDefault         = "INFO"
+	loggingLevelDescription     = "The level of the loggers of the application {'ALL', 'DEBUG', 'ERROR', 'INFO', 'WARN', 'OFF'}"
+)
+
+// JiraAPIResourceParameters is a struct that holds every possible parameters/flags known by the application.
+// They are all parsed via the flag.Parse() method of the Go flags api. The struct also contains the meta-parameters
+// (see meta-parametes.go).
+type JiraAPIResourceParameters struct {
+	JiraAPIUrl       *string
+	Username         *string
+	Password         *string
+	Context          Context
+	IssueList        []string
+	CustomFieldName  *string
+	CustomFieldType  *string
+	CustomFieldValue *string
+	LoggingLevel     *string
+
+	ActiveIssue string         // The **SINGLE** issue that the resource is currently processing
+	Meta        MetaParameters //
+	Flags       JiraAPIResourceFlags
+}
+
+// This struct is used to separate the parameters (which takes values in the command line) of the flags (which don't).
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [comment on exported type JiraAPIResourceFlags should be of the form "JiraAPIResourceFlags ..." (with optional leading article)](https://app.codacy.com/app/TurnsCoffeeIntoScripts/jira-api-resource/pullRequest?prid=4570750)